luna.c

```diff
@@ -370,7 +366,7 @@ int doccrypt(void *inout, long in_size) {
 		memcpy(&cbc_data, ivec, sizeof(DES_cblock));
 		DES_ecb3_encrypt(&cbc_data, &cbc_data, &ks1, &ks2, &ks3, DES_ENCRYPT);
  		for (i = 0; i < ((unsigned)in_size >= sizeof(DES_cblock) ? sizeof(DES_cblock) : (unsigned)in_size); i++) {
-			*(unsigned char*)inout++ ^= ((unsigned char*)cbc_data)[i];
+			*(unsigned char*)inout++ ^= ((unsigned char*)&cbc_data)[i];
```

Comment:
The name of the type is the same, but the actual underlying type isn't. This is how it should be now.